src/components/description_list/description_list_context.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { createContext } from 'react';
+import { EuiDescriptionListProps } from './description_list_types';
+
+interface DescriptionListContextProps {
+  align?: EuiDescriptionListProps['align'];
+  compressed?: EuiDescriptionListProps['compressed'];
+  textStyle?: EuiDescriptionListProps['textStyle'];
+  type?: EuiDescriptionListProps['type'];
+}
```

Comment:
Minor DRY suggestion
```suggestion
type DescriptionListContextProps = Partial<Pick<EuiDescriptionListProps, 'align' | 'compressed' | 'textStyle' | 'type'>> & { children?: React.ReactNode };
```

note: this will certainly need to be formatted correctly by Prettier, but this way you have just 1 declaration and can can lose the extra interface declaration with `&`